kernel/safe_typing.ml

```diff
@@ -515,8 +533,19 @@ let add_field ?(is_include=false) ((l,sfb) as field) gn senv =
     | SFBmodule mb, M -> Modops.add_module mb senv.env
     | _ -> assert false
   in
+  let sections = match sfb, gn with
+  | SFBconst cb, C con ->
+    let poly = Declareops.constant_is_polymorphic cb in
+    Section.push_constant ~poly con senv.sections
+  | SFBmind mib, I mind ->
+    let poly = Declareops.inductive_is_polymorphic mib in
+    Section.push_inductive ~poly mind senv.sections
+  | _, (M | MT) -> senv.sections
+  | _ -> assert false
+  in
```

Comment:
It seems pretty weird scoping-wise to have this information duplicated. Aren't regular environments already telescopes? It'd seem to me that section data should instead be stored in the regular env and discharge should proceed over there, right?

In particular I find very error prone and hard to understand that `Section.push_constant` is a no-op when no sections are opened.